DeviceTree: Convert 'generated_dts_board.h' into a source file

SebastianBoe opened the pull request:
extract_dts_includes.py has been generating DT output and then
concatenating it with fixup header files to create
'generated_dts_board.h'.

In this patch we instead introduce a source file named
'generated_dts_board.h' and have it \#include the appropriate DT
output and fixup files.

This results in a simpler system because users can now read
'generated_dts_board.h' as C source code to see how fixup files and
generated DT output relate to each other. Whereas before they would
have to either read documentation or python code to gain the same
understanding.

Also, it reduces the scope and complexity of one of our most bloated
python scripts, extract_dts_includes.py.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe committed:
DeviceTree: Convert 'generated_dts_board.h' into a source file

extract_dts_includes.py has been generating DT output and then
concatenating it with fixup header files to create
'generated_dts_board.h'.

In this patch we instead introduce a source file named
'generated_dts_board.h' and have it \#include the appropriate DT
output and fixup files.

This results in a simpler system because users can now read
'generated_dts_board.h' as C source code to see how fixup files and
generated DT output relate to each other. Whereas before they would
have to either read documentation or python code to gain the same
understanding.

Also, it reduces the scope and complexity of one of our most bloated
python scripts, extract_dts_includes.py.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

hakonfam commented on include/generated_dts_board.h:
```diff
@@ -0,0 +1,30 @@
+#include DTS_SHIELDS_FIXUP_FILE
+#endif
+
+#endif
```
Add comment matching checked define
```
#endif  /* DEVICE_TREE_BOARD_H */
```

hakonfam commented on include/generated_dts_board.h:
```diff
@@ -0,0 +1,30 @@
+ * Poorly named for legacy compatibility reasons.
+ */
+
+#ifndef DEVICE_TREE_BOARD_H
```
Define does not match file name.

SebastianBoe committed:
DeviceTree: Fix generated header comment

The header comment in 'generated_dts_board_unfixed.h' was including
'compatible' as a helpful text.

But something has been broken with how we extract 'compatible' because
it was being evaluated to 'q'. Giving a confusing header text:

Generated include file q

Since the mechanism is broken, and does not appear to be important, we
remove it.

Also, modernize how we generate multi-line strings.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

aurel32 commented:
Note that this commit also broke openthread support in Zephyr as OT is not compiled with the extra -D options.

SebastianBoe commented:
@pfalcon , @aurel32 : I'm working on fixing the regressions.
